lib/ansible/module_utils/basic.py

```diff
@@ -862,6 +867,12 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
                            ' Update the code for this module  In the future, AnsibleModule will'
                            ' always check for invalid arguments.', version='2.9')
 
+    def __del__(self):
+        try:
+            self._catch_warnings.__exit__(tuple())
```

Comment:
FTR in non-CPython implementations the time of when GC will call this may be inconsistent.

Reply:
Also, a tuple is not expected arg type: [https://docs.python.org/3/reference/datamodel.html#object.__exit__](https://docs.python.org/3/reference/datamodel.html#object.__exit__)

> If the context was exited without an exception, all three arguments will be None.


```suggestion
            self._catch_warnings.__exit__(None, None, None)
```